doc/user/flowspec.rst

```diff
@@ -0,0 +1,330 @@
+both IPtables and IPRule contexts.
+
+.. code-block:: shell
+
+   # iptables -t mangle --list match0x271ce00 -v
+   Chain match0x271ce00 (1 references)
+   pkts bytes target     prot opt in     out     source               destination
+   1700K  158M MARK       all  --  any    any     anywhere             anywhere
+        MARK set 0x101
+   1700K  158M ACCEPT     all  --  any    any     anywhere             anywhere
+
+   # ip rule list
+   0:from all lookup local
+   0:from all fwmark 0x101 lookup 257
+   32766:from all lookup main
+   32767:from all lookup default
+
+This allows us to see where the traffic is forwarded to.
+
+Limitations / Known issues
+--------------------------
+
+As you can see, Flowspec is rich and can be very complex.
+As of today, not all flowspec rules will be able to be converted into Policy
+based routing actions.
+
+- The `Linux Netfilter` driver is not integrated into FRR yet. Not having this piece
+of code prevents from injecting flowspec entries into the underlying system.
+
+- There are some limitations around filtering contexts
+
+If I take example of UDP ports, or TCP ports in flowspec, the information
+can be a range of ports, or a unique value. This case is handled.
+However, complexity can be increased, if the flow is a combination of a list of
+range of ports and an enumerate of unique values. Here this case is not handled.
+Similarly, it is not possible to create a filter for both src port and dst port.
+For instance, filter on src port from [1-1000] and dst port = 80.
+The same kind of complexity is not possible for packet length, ICMP type, ICMP code.
+
+- The first version of FRR Flowspec only handles a subset of Filter rules.
+
+It operates policy based routing, on the following Flowspec criteria : IP Address,
+UDP port or TCP port. The next implementation plans to provide the following : support for
+ICMP type/code , TCP flags, fragmentation and packet length.
+
+There are some other known issues:
+
+- The validation procedure depicted in RFC5575 is not available.
+
+This validation procedure has not been implemented, as this feature was not used
+in the existing setups you shared wih us.
+
+- The filtering action shaper value, if positive, is not used to apply shaping.
+
+If value is positive, the traffic is redirected to the wished destination,
+without any other action configured by Flowspec.
+It is recommended to configure Quality of Service if needed, more globally on
+a per interface basis.
+
+- upon crash or unknown event, *zebra* may not have time to flush pbr contexts.
+
+That is to say ipset/iptables and iprule contexts. This is also a consequence due
+to the fact that ip rule / ipset / iptables are not discovered at startup (not
+able to read appropriate contexts coming from BGP FS).
+
+Appendix
+--------
+
+More information with a public presentation that explains the design of Flowspec inside FRRouting.
+
+[Presentation]_
+
+.. [RFC5575] <https://tools.ietf.org/rfc/rfc5575.txt>
```

Comment:
You can remove this and just use Sphinx's `:rfc:` role as noted before.

Reply:
ok